### PR TITLE
feat: trino autocomplete

### PR DESCRIPTION
**Pull request by NikitaShkaruba**

### CODE REVIEW

File: src/autocomplete/databases/postgresql/grammar/PostgreSqlLexer.tokens
Comment (NikitaShkaruba) [inline src/autocomplete/databases/postgresql/grammar/PostgreSqlLexer.tokens:1]: This file got generated after I ran generate:postgresql

File: src/autocomplete/databases/trino/index.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import {
+    ExtractStatementPositionsResult,
+    extractStatementPositionsFromQuery,
+} from '../../shared/extract-statement-positions-from-query';
+import {TrinoParser} from './generated/TrinoParser';
+
+export interface TrinoAutocompleteResult extends SqlAutocompleteResult {
+    suggestViewsOrTables?: TableOrViewSuggestion;
+    suggestSchemas?: true;
+    suggestCatalogs?: true;
```
Comment (NikitaShkaruba) [inline src/autocomplete/databases/trino/index.ts:18]: These 3 fields are the ones i'm targeting in the first alpha version

File: src/autocomplete/databases/trino/grammar/TrinoLexer.tokens [hunk 2]
```diff
@@ -0,0 +1,665 @@
+ABSENT_=1
```
Comment (roberthovsepyan) [inline src/autocomplete/databases/trino/grammar/TrinoLexer.tokens:1]: yeah this is definitely a bug in generation
this file is already present in generated/ directory, it shouldn't be here

File: src/autocomplete/databases/trino/trino-autocomplete.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    let suggestViewsOrTables: TrinoAutocompleteResult['suggestViewsOrTables'];
+
+    for (const [ruleId, rule] of rules) {
+        if (!isStartingToWriteRule(cursorTokenIndex, rule)) {
+            continue;
+        }
+
+        switch (ruleId) {
+            case TrinoParser.RULE_identifier: {
+                suggestViewsOrTables = TableOrViewSuggestion.ALL;
```
Comment (roberthovsepyan) [inline src/autocomplete/databases/trino/trino-autocomplete.ts:73]: does this work? it seems like identifier is used everywhere

File: src/autocomplete/databases/trino/grammar/TrinoLexer.tokens (hunk 2 above)
Comment (NikitaShkaruba) [reply src/autocomplete/databases/trino/grammar/TrinoLexer.tokens:1]: I've removed it, thanks

File: src/autocomplete/databases/trino/trino-autocomplete.ts (hunk 3 above)
Comment (NikitaShkaruba) [reply src/autocomplete/databases/trino/trino-autocomplete.ts:73]: Currently, No. But I'll fix it in the next PR

**Review by roberthovsepyan — APPROVED** [review]

**Pull request merged**